pkg/controller/registrysyncer/registrysyncer_test.go

```diff
@@ -0,0 +1,454 @@
+			appCIClient: fakeclient.NewFakeClientWithScheme(scheme.Scheme, []runtime.Object{applyconfigISTagNewerSameName.DeepCopy(), applyconfigIS.DeepCopy()}...),
+
+			verify: func(apiCIClient ctrlruntimeclient.Client, appCIClient ctrlruntimeclient.Client) error {
+				for clusterName, client := range map[string]ctrlruntimeclient.Client{apiCI: apiCIClient, appCI: appCIClient} {
+					actualImageStreamImport := &imagev1.ImageStreamImport{}
+					err := client.Get(ctx, types.NamespacedName{Name: "applyconfig", Namespace: "ci"}, actualImageStreamImport)
+					if !errors.IsNotFound(err) {
+						return fmt.Errorf("unexpected import on %s", clusterName)
+					}
+				}
+				return nil
+			},
+		},
+		{
+			name: "import check failed",
+			request: types.NamespacedName{
+				Name:      "applyconfig:latest",
+				Namespace: "ci",
+			},
+			apiCIClient: fakeclient.NewFakeClientWithScheme(scheme.Scheme, []runtime.Object{applyconfigISTag.DeepCopy(), applyconfigIS.DeepCopy()}...),
+			appCIClient: bcc(fakeclient.NewFakeClientWithScheme(scheme.Scheme), func(c *imageImportStatusSettingClient) { c.failure = true }),
+			expected:    fmt.Errorf("imageStreamImport did not succeed: reason: , message: failing as requested"),
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			r := &reconciler{
+				ctx: context.Background(),
+				log: logrus.NewEntry(logrus.New()),
+				registryClients: map[string]ctrlruntimeclient.Client{
+					apiCI: tc.apiCIClient,
+					appCI: tc.appCIClient,
+				},
+				pullSecretGetter: pullSecretGetter,
+			}
+
+			request := reconcile.Request{NamespacedName: tc.request}
+			actual := r.reconcile(request, r.log)
+
+			if diff := cmp.Diff(tc.expected, actual, equateErrorMessage); diff != "" {
+				t.Errorf("actualError does not match expectedError, diff: %s", diff)
+			}
+			if actual == nil && tc.verify != nil {
+				if err := tc.verify(tc.apiCIClient, tc.appCIClient); err != nil {
+					t.Errorf("unexpcected error: %v", err)
+				}
+			}
+		})
+	}
+}
+
+func bcc(upstream ctrlruntimeclient.Client, opts ...func(*imageImportStatusSettingClient)) ctrlruntimeclient.Client {
+	c := &imageImportStatusSettingClient{
+		Client: upstream,
+	}
+	for _, opt := range opts {
+		opt(c)
+	}
+	return c
+}
+
+type imageImportStatusSettingClient struct {
+	ctrlruntimeclient.Client
+	failure bool
+}
+
+func (client *imageImportStatusSettingClient) Create(ctx context.Context, obj runtime.Object, opts ...ctrlruntimeclient.CreateOption) error {
+	if asserted, match := obj.(*imagev1.ImageStreamImport); match {
+		asserted.Status.Images = []imagev1.ImageImportStatus{{}}
+		if client.failure {
+			asserted.Status.Images[0].Status.Message = "failing as requested"
+		} else {
+			asserted.Status.Images[0].Image = &imagev1.Image{}
+		}
+	}
+	return client.Client.Create(ctx, obj, opts...)
+}
+
+var (
+	//https://github.com/google/go-cmp/issues/24#issuecomment-317635190
+	equateErrorMessage = cmp.Comparer(func(x, y error) bool {
```

Comment:
Can you move this into `pkg/testhelper`? I know for sure that I want to use it in the future as well :)

Reply:
Done.